src/Core/src/Handlers/Image/ImageHandler.Windows.cs

```diff
@@ -55,6 +57,29 @@ protected override void RemoveContainer()
 			UpdateValue(nameof(IView.Width));
 		}
 
+		/// <inheritdoc/>
+		public override Graphics.Size GetDesiredSize(double widthConstraint, double heightConstraint)
+		{
+			// Compute a possible size without mutating platform properties during measure.
+			var possibleSize = base.GetDesiredSize(widthConstraint, heightConstraint);
+
+			// For AspectFit + non-Fill alignment, don't exceed intrinsic bitmap size so alignment works.
+			if (VirtualView.Aspect == Aspect.AspectFit
```

Comment:
Current logic only caps if both horizontal and vertical alignments are non-Fill. This might not handle cases where one is Fill and the other isn't optimally.

Example: HorizontalLayoutAlignment = Center, VerticalLayoutAlignment = Fill.

With AspectFit, the image should fill the height (scaling width accordingly), but allow centering horizontally if the scaled width < container width. Without capping width, it might force-fill horizontally too, ignoring the center alignment.

Should cap width if Horizontal != Fill, cap height if Vertical != Fill?